docs/madr/decisions/055-default-hostname-generators.md

```diff
@@ -0,0 +1,218 @@
+# Default HostnameGenerators
+
+* Status: accepted
+
+## Context and Problem Statement
+
+We introduced HostnameGenerator for MeshService and MeshExternalService which is also compatible with upcoming MeshMultizoneService.
+This MADR proposes default generators taking into account different scenarios.
+
+## Considered Options
+
+Template and matching:
+* Proposed options
+
+Shipping:
+* Document
+* Create
+
+Suffix:
+* `.svc.{{ type }}.{{ zone }}`
+* `.{{ zone }}.{{ type }}`
+
+## Decision Outcome
+
+Template and matching - Proposed options.
+Shipping - short term document. Once we get all prerequisites and MeshService/MeshExternalService/MeshMultizoneService goes GA, we create by default.
```

Comment:
When we say shipping is this just resources added to the helm chart?
If yes:
- what's the story for universal?
If no:
- if it's code generated we must make sure people can start in prod with nothing and define everything with IaaC

Reply:
I think it should be the same mechanism as for creating any other "global-scoped" resources so:
* Default Mesh
* Default RBAC in Kong Mesh

so not in HELM but in the code with option to opt-out

Reply:
OK

We need to have a better way to start empty and have customers create these in the future but that's parallel to this MADR